src/background/chrome-api.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * This module provides a wrapper around the Chrome / WebExtension APIs that
+ * the extension uses.
+ *
+ * It has several purposes:
+ *  - Provide a Promise-returning interface for all async APIs
+ *  - Abstract over the differences between different browsers
+ *  - Provide a seam that can be easily mocked in tests
+ */
+
+/**
+ * Wrap the browser APIs exposed via the `chrome` object to return promises.
+ *
+ * This is exposed for testing. Consumers should use {@link chromeAPI}.
+ */
+export function getChromeAPI(chrome = globalThis.chrome) {
+  // In the test environment, the `chrome` global may not exist. This is ignored
+  // for the purposes of determining the return type.
+  if (typeof chrome === 'undefined') {
+    return /** @type {never} */ (null);
+  }
+
+  /**
+   * @template Result
+   * @typedef {(r: Result) => void} Callback
+   */
+
+  /**
+   * Convert an async callback-accepting Chrome API to a Promise-returning version.
+   *
+   * TypeScript may complain if the API has a Manifest V3-only overload that
+   * returns a Promise. Use {@link promisifyAlt} as a workaround.
+   *
+   * @template {any[]} Args
+   * @template Result
+   * @param {(...args: [...Args, Callback<Result>]) => void} fn
+   * @return {(...args: Args) => Promise<Result>}
+   */
+  const promisify =
+    fn =>
+    (...args) => {
+      return new Promise((resolve, reject) => {
+        fn(...args, (/** @type {Result} */ result) => {
+          const lastError = chrome.runtime.lastError;
```

Comment:
Given that we're in async-land, can we be confident that `lastError` is cogent to this action?

Reply:
Yes, as far as I know. This is the idiomatic way to check for errors according to [examples in the Chrome docs](https://developer.chrome.com/docs/extensions/mv3/promises/#standard-callback-implementation).